pkg/apis/kops/channel.go

```diff
@@ -352,3 +364,30 @@ func (c *Channel) HasUpstreamImagePrefix(image string) bool {
 		strings.HasPrefix(image, "ubuntu-os-cloud/ubuntu-2004-focal-") ||
 		strings.HasPrefix(image, "Canonical:0001-com-ubuntu-server-focal:20_04-lts-gen2:")
 }
+
+// GetPackageVersion returns the version for the package, or an error if could not be found.
+func (c *Channel) GetPackageVersion(name string) (*util.Version, error) {
+	var matches []*PackageVersionSpec
+
+	for i := range c.Spec.Packages {
+		pkg := &c.Spec.Packages[i]
+		if pkg.Name != name {
+			continue
+		}
+
+		matches = append(matches, pkg)
+	}
+
+	if len(matches) == 0 {
+		return nil, fmt.Errorf("found no packages in channel for name=%q", name)
+	}
+
+	if len(matches) != 1 {
+		return nil, fmt.Errorf("found multiple packages in channel for name=%q", name)
```

Comment:
Is the intent that we'll only ever have one package per addon? what if we end up needing to support per-k8s-version packages, similar to bootstrapchannelbuilder now?

Reply:
Good point.  So the channel has two versions per "bundled" addon ... the operator version, and the version we should inject into the CRD.   The operator version, yes, I expect that we will resolve that to a single version.  We don't currently have it, but we could have kubernetesVersion selectors and/or kopsVersion selectors in the channels file.  There's probably a good case for adding per-k8s-version now; ideally these operators wouldn't be specific to kOps versions though so I'm not sure of the case for a per-kops version.

Reply:
I added filtering support for kOps & k8s version - it's much hard to add later.  Thanks for the suggestion @rifelpet